vault/raft.go

```diff
@@ -694,7 +694,10 @@ func (c *Core) raftCreateTLSKeyring(ctx context.Context) (*raft.TLSKeyring, erro
 	}
 
 	if raftTLSEntry != nil {
-		return nil, fmt.Errorf("TLS keyring already present")
+		// If resetTLSKeyring is set, we will overwrite the existing keyring.
+		if !resetTLSKeyring {
```

Comment:
Instead of this check, we could do something like this:
```
	if _, usingRaftStorage := c.underlyingPhysical.(*raft.RaftBackend); usingRaftStorage {
		`return nil, fmt.Errorf("TLS keyring already present")`
	}
```

This avoids the need to add a new parameter, and I think is a lot cleaner. The user shouldn't need to provide a boolean if we can figure it out ourselves